helmfile/values/fluentd-sc.yaml.gotmpl

```diff
@@ -1,196 +1,48 @@
-      key_name log
-      reserve_data true
-      reserve_time true #This is the line that is changed from the default config
-      remove_key_name_field true
-      <parse>
-        @type multi_format
-        <pattern>
-          format json
-        </pattern>
-        <pattern>
-          format none
-        </pattern>
-      </parse>
-    </filter>
-
-    # count number of incoming records per tag
-    <filter **>
-      @type prometheus
-      <metric>
-        name fluentd_input_status_num_records_total
-        type counter
-        desc The total number of incoming records
-        <labels>
-          tag ${tag}
-          hostname ${hostname}
-        </labels>
-      </metric>
-    </filter>
-
-  output.conf: |-
-    <match **>
-      @id output-forwarding
-      @type forward
-      send_timeout 60s
-      recover_wait 10s
-      hard_timeout 60s
-      <server>
-        name aggregator
-        host fluentd-aggregator
-        port 24224
-        weight 60
-      </server>
-      #<secondary>
-      #  @type file
-      #  path /var/log/fluentd/forward-failed
-      #</secondary>
-      <buffer>
-        @type file
-        path /var/log/fluentd-buffers
-        flush_mode interval
-        retry_type exponential_backoff
-        flush_thread_count 2
-        flush_interval 5s
-        retry_forever
-        retry_max_interval 30
-        chunk_limit_size "#{ENV['OUTPUT_BUFFER_CHUNK_LIMIT']}"
-        queue_limit_length "#{ENV['OUTPUT_BUFFER_QUEUE_LIMIT']}"
-        overflow_action block
-      </buffer>
-    </match>
-    <match **>
-      @type prometheus
-      <metric>
-        name fluentd_output_status_num_records_total
-        type counter
-        desc The total number of outgoing records
-        <labels>
-          tag ${tag}
-          hostname ${hostname}
-        </labels>
-      </metric>
-    </match>
+#log aggregator are daemons that continuously receives logs from log forwarders and push the log to external storage (in our case s3)
+aggregator:
+  configMap: fluentd-aggregator-configmap
+
+  replicaCount: 1
+
+  # Persist buffer so that the last chunks of logs can be retrieved in case of a
+  # disaster.
+  persistence:
+    enabled: true
+    size: 10Gi
+
+  extraEnv:
+   - name: AWS_ACCESS_KEY_ID
+     valueFrom:
+       secretKeyRef:
+         name: s3-credentials
+         key: s3_access_key
+   - name: AWS_ACCESS_SECRET_KEY
+     valueFrom:
+       secretKeyRef:
+         name: s3-credentials
+         key: s3_secret_key
+
+#log forwarders are daemons that are installed on every node to collect local events and forward these events to the log aggregator(s).)
+# see https://docs.fluentd.org/deployment/high-availability
+forwarder:
+  configMap: fluentd-forwarder-configmap
+
+  extraEnv:
+    - name: OUTPUT_BUFFER_CHUNK_LIMIT
+      value: "8"
+    - name: OUTPUT_BUFFER_QUEUE_LIMIT
+      value: "4096"
 
 tolerations:
 - effect: NoSchedule
   key: "node-role.kubernetes.io/master"
   value: ""
```

Comment:
This should not be here, but it should be moved to the sc.config.
Right now it's merged with what the users specifies, see line 3.

Reply:
Forgot - you have to make it so that we can configure tolerations, affinity, resources etc for both the aggeragator and forwarders.